--- ./crypto/ocalls.h	2018-04-18 17:42:17.053424174 +0100
+++ ../libseal/./crypto/ocalls.h	2018-04-18 17:30:58.232831774 +0100
@@ -51,6 +51,8 @@
 uid_t ocall_getuid();
 int ocall_unlink(const char* pathname);
 int ocall_fsync(int fd);
+ssize_t ocall_pread(int fd, void *buf, size_t count, off_t offset);
+ssize_t ocall_pwrite(int fd, const void *buf, size_t count, off_t offset);
 
 void ocall_execute_ssl_ctx_info_callback(const SSL *ssl, int type, int val, void *cb);
 int ocall_alpn_select_cb(SSL *s, const unsigned char **out, unsigned char *outlen, const unsigned char *in, unsigned int inlen, void *arg, void *cb);
